cmd/agent/install_mac_os.sh

```diff
@@ -365,6 +353,22 @@ function plist_modify_user_group() {
     $sudo_cmd sh -c "sed $i_cmd -e \"${closing_dict_line},${closing_dict_line}s|</dict>|<key>$user_parameter</key><string>$user_value</string>\n</dict>|\" -e \"${closing_dict_line},${closing_dict_line}s|</dict>|<key>$group_parameter</key><string>$group_value</string>\n</dict>|\" \"$plist_file\""
 }
 
+# Determine agent flavor to install
+if [ -z "$agent_dist_channel" ]; then
+    dmg_url_prefix="$dmg_base_url/datadog-agent-${dmg_version}"
+else
+    dmg_url_prefix="$dmg_base_url/$agent_dist_channel/datadog-agent-${dmg_version}"
+fi
+
+dmg_url="$dmg_url_prefix.$arch.dmg"  # favor architecture-specific DMG, if available
+if [ "$(curl --head --output /dev/null --silent --write-out '%{http_code}' "$dmg_url")" = 404 ]; then
+    dmg_url="$dmg_url_prefix.dmg"  # fallback to "universal" DMG
+    if [ "$arch" = arm64 ] && ! /usr/bin/pgrep oahd >/dev/null 2>&1; then
+        printf "\033[31mRosetta is needed to run datadog-agent on $arch.\nYou can install it by running the following command :\n/usr/sbin/softwareupdate --install-rosetta --agree-to-license\033[0m\n"
+        exit 1
+    fi
```

Comment:
Note to reviewers: snippet moved from somewhere above so that it's executed only when applicable.
In particular, if `$arch` is `arm64` and the retained DMG is _also_ `arm64`, there's no point in asking for Rosetta 2 installation.